docs/topics/articles.md

```diff
@@ -0,0 +1,184 @@
+# `Article`s and `ListingPage`s
+
+`Article`s are publishable content types that live in collections. Normally
+articles are mounted under a parent's URL.
+
+An example of an article is a Press Release, which would be mounted under a
+Press Release Listing Page. The Listing Page has the URL "/press" and thus all
+the Press Releases have the URL "/press/<slug>".
+
+## `ArticleBase`
+
+The `icekit.articles.models.ArticleBase` inherits `title`, `slug` and
+publishing fields, and each `ArticleBase` subclass implements some key
+functionality:
+
+1. **Links to parents.**
+In order for an Article to know its URL (which is based on a parent's URL), it
+should define a `parent` link, which is normally a link to a `ListingPage`.
```

Comment:
Back to first principles. The need is to allow objects to appear at some url, such as "/watch/<slug>", where "watch" is a grouping term chosen in some way by the user. There are two candidate ways of doing this: assign a category, or assign a page.

If you need a page at /watch/ that lists the items, as we often do, then it seems convenient to create such a page, and link the items to it. That's what this pattern is.

If you don't need a page, you don't have to have one; you make "parent" link to a category model and use a traditional URL pattern.

If you like the listing page, but don't like the way items have a parent to it, you don't need to use the parent pattern.

If you need a page that filters items in a different way, then either extend/upgrade the listing page model or implement something else. No dramas.

The only time we need to use this pattern is when it is architecturally sensible: when there is content to be listed, and we need one or more pages to list them under. I don't think we're painting ourselves into a corner.

(I was sat with the team at SFMOMA, and I can tell you the page+category approach was opaque enough that I needed to do the setup for them every time. I myself had to sift through the code the first time to figure out what was going on. They expected to be able to see the article once they had attached the category, which didn't work, because the URL pattern for the article is defined by the Page.)